src/access/AccessRegistry.sol

```diff
@@ -0,0 +1,160 @@
+        factoryVersion++;
+        emit AccessPointFactoryUpgraded(beacon.implementation(), address(newImpl));
+        beacon.upgradeTo(address(newImpl));
+    }
+
+    /**
+     * @dev Authorize the upgrade. Only by an owner.
+     * @param newImplementation address of the new implementation
+     */
+    // This function is called by the proxy contract when the factory is upgraded
+    function _authorizeUpgrade(address newImplementation) internal view override onlyOwner {
+        (newImplementation);
+    }
+
+    /* ============ View Functions ============ */
+
+    /// @inheritdoc IAccessRegistry
+    function isWorkflowAllowed(address workflow) external view returns (bool) {
+        return _workflows[workflow];
+    }
+
+    /// @inheritdoc IAccessRegistry
+    function getAccessPoint(address user) external view returns (IAccessPoint accessPoint) {
+        accessPoint = _accessPoints[user];
+    }
+
+    /// @inheritdoc IAccessRegistry
+    function getAddress(address owner) public view override returns (address) {
+        return Create2.computeAddress(
+            bytes32(abi.encodePacked(owner)),
+            keccak256(
+                abi.encodePacked(
+                    type(SafeBeaconProxy).creationCode,
+                    abi.encode(address(beacon), abi.encodeCall(IAccessPoint.initialize, (owner)))
+                )
+            )
+        );
+    }
+
+    /// @dev TODO:Should be removed. Added because Kinto EntryPoint needs this function.
+    function getWalletTimestamp(address) external pure returns (uint256) {
+        return 1;
+    }
+
+    /* ============ State Change ============ */
+
+    /// @inheritdoc IAccessRegistry
+    function disallowWorkflow(address workflow) external onlyOwner {
+        _workflows[workflow] = false;
+        emit WorkflowStatusChanged(workflow, false);
+    }
+
+    /// @inheritdoc IAccessRegistry
+    function allowWorkflow(address workflow) external onlyOwner {
+        _workflows[workflow] = true;
+        emit WorkflowStatusChanged(workflow, true);
+    }
+
+    /// @inheritdoc IAccessRegistry
+    function deployFor(address user)
+        external
+        override
+        onlyNonAccessPointOwner(user)
+        returns (IAccessPoint accessPoint)
+    {
+        return _deploy(user);
+    }
+
+    /* ============ Internal ============ */
+
+    /// @dev See `deploy`.
+    function _deploy(address owner) internal returns (IAccessPoint accessPoint) {
+        // Use the address of the owner as the CREATE2 salt.
+        bytes32 salt = bytes32(abi.encodePacked(owner));
+
+        // Deploy the accessPoint with CREATE2.
+        accessPoint = IAccessPoint(
+            payable(new SafeBeaconProxy{salt: salt}(address(beacon), abi.encodeCall(IAccessPoint.initialize, (owner))))
+        );
+
+        // Associate the owner and the accessPoint.
+        _accessPoints[owner] = accessPoint;
+
```

Comment:
should we allow some workflows by default?

Reply:
Yes. I was thinking the same. But a better way to do it is to just inherit them by `AccessPoint` so people can call them directly and no need to check if they enabled/disabled. What do you think?